taichi/program/async_engine.cpp

```diff
@@ -208,7 +208,9 @@ void AsyncEngine::launch(Kernel *kernel, Context &context) {
 void AsyncEngine::synchronize() {
   TI_AUTO_PROF;
   bool modified = true;
-  sfg->rebuild_graph(/*sort=*/false);
+  sfg->reid_nodes();
+  sfg->reid_pending_nodes();
+  sfg->sort_node_edges();
```

Comment:
I reverted the `rebuild_graph()` and added a specific `sort_node_edges` as suggested by @xumingkuan .